### PR TITLE
add eth_BlockNumber

### PR DESCRIPTION
**Pull request by ayrat555**
resolves https://github.com/poanetwork/blockscout/issues/1919

```curl -X POST  "http://localhost:4000/api?module=block&action=eth_block_number&id=10"```

### CODE REVIEW

**Comment by zachdaniel** [comment]
Isn't there a block number cache that keeps.track of the max block number, since max block number is an expensive query?

**Comment by ayrat555** [comment]
@zachdaniel yes, there is. Is it ok if a user will get not up-to-date value? @vbaranov

**Comment by zachdaniel** [comment]
I thought the cache was kept always up to date as new blocks came in.

**Comment by vbaranov** [comment]
@ayrat555 Zach's point is correct. We should return every time up-to-date data.

**Comment by ayrat555** [comment]
cache is updated every 10 minutes

**Comment by vbaranov** [comment]
@ayrat555 can we trigger a cache update for this endpoint not by the scheduler, but with every new block coming?

**Comment by ayrat555** [comment]
@vbaranov yes should I add it in this PR?

**Comment by igorbarinov** [comment]
the cache should be invalidated with each new block coming to the db

**Comment by igorbarinov** [comment]
otherwise, it makes no sense to use it

**Comment by zachdaniel** [comment]
I think keeping the data up to date would be pretty easy, since every consensus block would just send that cache a message, and if the number was lower than that consensus block number the number would become that block number. Then we’d still double check every 10 minutes to catch any misalignment.

**Comment by igorbarinov** [comment]
btw in a week, we had `40,066,512` requests of this type on xDai RPC
we need to cache it for sure

**Comment by ayrat555** [comment]
@vbaranov @zachdaniel I changed cache so now it will start updating after getting a new highest block in realtime fetcher. And `eth_BlockNumber` uses this cache

File: apps/explorer/lib/explorer/chain.ex [hunk 1]
```diff
@@ -1059,6 +1059,22 @@ defmodule Explorer.Chain do
     end
   end
 
+  @spec fetch_max_block_number() :: non_neg_integer
```
Comment (zachdaniel) [inline apps/explorer/lib/explorer/chain.ex]: We can probably remove this now, right?

File: apps/explorer/lib/explorer/chain/block_number_cache.ex [hunk 2]
```diff
@@ -59,6 +59,16 @@ defmodule Explorer.Chain.BlockNumberCache do
     end
   end
 
+  def update_max_number(max) do
```
Comment (zachdaniel) [inline apps/explorer/lib/explorer/chain/block_number_cache.ex]: We should probably just update either number, with any new block. Either the max if this number is higher than that, or the min if this number is lower than that.

File: apps/explorer/lib/explorer/chain/block_number_cache.ex [hunk 3]
```diff
@@ -59,6 +59,23 @@ defmodule Explorer.Chain.BlockNumberCache do
     end
   end
 
+  def update(number) do
```
Comment (saneery) [inline apps/explorer/lib/explorer/chain/block_number_cache.ex]: I think here is need a spec

**Comment by ayrat555** [comment]
@vbaranov done

**Comment by vbaranov** [comment]
@ayrat555 in addition,

1. see the difference of the returned response from the created API endpoint:
```
viktor$ curl -X POST -H "Content-type: application/json" --data '{"jsonrpc":"2.0","method":"eth_blockNumber","params":[],"id":1}' "http://localhost:4000/api?module=block&action=eth_block_number"
{"result":"37DAE1","jsonrpc":"2.0","id":1}viktor$
```


 and API from the node:

```
viktor$ curl -X POST -H "Content-type: application/json" --data '{"jsonrpc":"2.0","method":"eth_blockNumber","params":[],"id":1}' "https://dai.poa.network"
{"jsonrpc":"2.0","result":"0x37dbc0","id":1}
```

Differences:
a. Wrong order of properties in JSON
b. There is no line break at the end of the response

2. As we created standard JSON RPC endpoint and it is publicly accessible, we should list in the `/api_docs` page

**Comment by ayrat555** [comment]
@vbaranov  I addressed all your points except for ` Wrong order of properties in JSON`. It is hash and it does not specify order

**Comment by igorbarinov** [comment]
in JSON spec order is unordered. Although, having order will make it easier to read/use/debug. All ethereum clients preserve order. Let's preserve order too.

```
 An object is an unordered collection of zero or more name/value
   pairs, where a name is a string and a value is a string, number,
   boolean, null, object, or array.

   An array is an ordered sequence of zero or more values.
```

**Comment by ayrat555** [comment]
I added order

File: apps/explorer/lib/explorer/chain/block_number_cache.ex [hunk 4]
```diff
@@ -59,6 +59,24 @@ defmodule Explorer.Chain.BlockNumberCache do
     end
   end
 
+  @spec update(non_neg_integer()) :: boolean()
+  def update(number) do
```
Comment (pasqu4le) [inline apps/explorer/lib/explorer/chain/block_number_cache.ex]: If this is now called for every incoming block, doesn't it make the time-passed check in the `value` function just above (and everything `cache_period` related as well) irrelevant?

`update_cache` still gets called in `setup` after all

Comment (ayrat555) [reply apps/explorer/lib/explorer/chain/block_number_cache.ex]: @pasqu4le sounds reasonable. I removed it

**Comment by vbaranov** [comment]
@ayrat555 the block number correctly updates now and the order is correct. Let's make the change in the response representation
```
viktor$ curl -X POST -H "Content-type: application/json" --data '{"jsonrpc":"2.0","method":"eth_blockNumber","params":[],"id":1}' "https://dai.poa.network"
{"jsonrpc":"2.0","result":"0x381c89","id":1}
viktor$ curl -X POST -H "Content-type: application/json" --data '{"jsonrpc":"2.0","method":"eth_blockNumber","params":[],"id":1}' "http://localhost:4000/api?module=block&action=eth_block_number"
{
"jsonrpc": "2.0",
"result": "0x381c89",
"id": 1
}
```

Let's minify the response from 

```
{
"jsonrpc": "2.0",
"result": "0x381c89",
"id": 1
}
```

to

`{"jsonrpc":"2.0","result":"0x381c89","id":1}`

Thus, it should be
1. in one line
2. no whitespaces
3. In the end, it should be one line-break, not two as for now

**Comment by ayrat555** [comment]
@vbaranov done

**Pull request merged**